test/simple.test.ts

```diff
@@ -263,3 +264,22 @@ test('workerId should never be duplicated', async () => {
   await pool.destroy()
   await sleep(5000)
 }, 30000)
+
+/* test('isolateWorkers: true with minThreads of 0 should not halt(#42)', async () => {
+  const minThreads = 0,
```

Comment:
@overlookmotel see this bit, it's 0 but it does not halt.

Reply:
Thanks for working on this. I'll try to put together a failing test. I definitely saw this halting behavior in "real world". But give me a few days - work is crazy at the moment.

Reply:
Ah I think know what it is. This bug occurs when you have tasks waiting in the queue when a task completes. It never drains the queue.

If you add `maxThreads + 1` tasks to the queue synchronously, then you'll see the last task never gets run.

```js
const promises = [];
for (let i = 0; i < maxThreads + 1; i++) {
  promises.push( await pool.run({}) );
}
const promiseForAll = Promise.all(promises);
await promiseForAll; // `promiseForAll` never resolves
```

I just tried to create a PR with failing test, but I've already forked `piscina` so Github won't let me fork `tinypool` too - you can't fork a fork as well as the origin, it seems.